src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -558,3 +558,32 @@ case object RETURN extends OpCode(0xf3, 2, 0, G_zero) {
     calcMemCost(state.memory.size, addr, size)
   }
 }
+
+case object SELFDESTRUCT extends OpCode(0xff, 1, 0, G_selfdestruct) {
+  protected def exec(state: ProgramState): ProgramState = {
+    val (refundDW, stack1) = state.stack.pop
+    val refundAddr: Address = Address(refundDW)
+    val ownerAccount: Account = state.context.account
+    val ownerAddr: Address = state.context.env.ownerAddr
+    val gasRefund: BigInt = if (state.addressesToDelete contains ownerAddr) 0 else R_selfdestruct
```

Comment:
Actually doing according to YP the code and storage should remain intact until transaction finalisation. But there are 2 other problems for subsequent calls to this account within the same transaction:

- account's balance update will not be seen on subsequent call
- a subsequent self-destruction crediting a previously inexistent account will cost more gas than it should